src/H5.c

```diff
@@ -257,26 +253,32 @@ H5_init_library(void)
      * The dataspace interface needs to be initialized so that future IDs for
      *   dataspaces work.
      */
-    if (H5E_init() < 0)
-        HGOTO_ERROR(H5E_FUNC, H5E_CANTINIT, FAIL, "unable to initialize error interface")
-    if (H5VL_init_phase1() < 0)
-        HGOTO_ERROR(H5E_FUNC, H5E_CANTINIT, FAIL, "unable to initialize vol interface")
-    if (H5P_init_phase1() < 0)
-        HGOTO_ERROR(H5E_FUNC, H5E_CANTINIT, FAIL, "unable to initialize property list interface")
-    if (H5AC_init() < 0)
-        HGOTO_ERROR(H5E_FUNC, H5E_CANTINIT, FAIL, "unable to initialize metadata caching interface")
-    if (H5L_init() < 0)
-        HGOTO_ERROR(H5E_FUNC, H5E_CANTINIT, FAIL, "unable to initialize link interface")
-    if (H5FS_init() < 0)
-        HGOTO_ERROR(H5E_FUNC, H5E_CANTINIT, FAIL, "unable to initialize FS interface")
-    if (H5S_init() < 0)
-        HGOTO_ERROR(H5E_FUNC, H5E_CANTINIT, FAIL, "unable to initialize dataspace interface")
-
+    /* clang-format off */
+    struct {
+        herr_t (*func)(void);
+        const char *descr;
```

Comment:
It would be nice I think to have a macro like you did for TERMINATOR that combines both phases of the initialization, e.g., H5VL_init_phase1 and H5VL_init_phase2 for "VOL"

Reply:
I don't understand what you mean.

Reply:
Just meant to have 2 funcs instead of one and a macro that sets the phase1/phase2 but not a big deal, just a suggestion. Just so that you have only one line for each module instead of having it decomposed.

Reply:
Let's revisit this later if more phased initializations crop up.